src/main/java/com/hcl/appscan/cli/handlers/InvokeDynamicScan.java

```diff
@@ -278,11 +283,17 @@ private  Optional<ScanResults> runScanAndGetResults() throws Exception {
 
```

Comment:
The method runScanAndGetResults() seems to be doing too much work. My suggestion would be to split this into at least 2 separate functions. One for executing the scan and another for getting the results. That will make the code easier read and easier to maintain moving forward.
One other suggestion is to use specific Exception types whenever possible, as opposed to throwing and catching the base Exception type.

Reply:
Sure , Matt . Agreed. I have now break down the method and created a seperate one for processing the results.
Will commit the changes. I think I can break down the method further , but for the sake of release deadline , I can do some more code refactoring in later tasks for future releases.